src/Treeview.ts

```diff
@@ -262,42 +266,40 @@ export class JsonOutlineProvider implements vscode.TreeDataProvider<string> {
+            if (keyOrResourceNode.toString().toUpperCase() === "handler".toUpperCase()) { icon = "label.svg" }
+            if (keyOrResourceNode.toString().toUpperCase() === "parameters".toUpperCase()) { icon = "parameters.svg" }
+            if (keyOrResourceNode.toString().toUpperCase() === "variables".toUpperCase()) { icon = "variables.svg" }
+            if (keyOrResourceNode.toString().toUpperCase() === "resources".toUpperCase()) { icon = "resources.svg" }
+            if (keyOrResourceNode.toString().toUpperCase() === "outputs".toUpperCase()) { icon = "outputs.svg" }
         }
-        // Is current element a element of a root element?
+        // Is current element an element of a root element?
         else if (elementInfo.current.level === 2) {
             // Get root value
             const rootNode = this.tree.getValueAtCharacterIndex(elementInfo.root.key.start);
 
-            if (rootNode._value.toUpperCase() === "parameters".toUpperCase()) { icon = "parameters.svg" }
-            if (rootNode._value.toUpperCase() === "variables".toUpperCase()) { icon = "variables.svg" }
-            if (rootNode._value.toUpperCase() === "outputs".toUpperCase()) { icon = "outputs.svg" }
+            if (rootNode.toString().toUpperCase() === "parameters".toUpperCase()) { icon = "parameters.svg" }
+            if (rootNode.toString().toUpperCase() === "variables".toUpperCase()) { icon = "variables.svg" }
+            if (rootNode.toString().toUpperCase() === "outputs".toUpperCase()) { icon = "outputs.svg" }
         }
 
-        // If resourceType element is found on resource objects set to specific resourceType Icon or else a a default resource icon
-        if (elementInfo.current.level > 1 && elementInfo.current.key.type === "ObjectValue") {
+        // If resourceType element is found on resource objects set to specific resourceType Icon or else a default resource icon
+        if (elementInfo.current.level > 1 && elementInfo.current.key.kind === Json.ValueKind.ObjectValue) {
             const rootNode = this.tree.getValueAtCharacterIndex(elementInfo.root.key.start);
 
-            if (rootNode._value.toUpperCase() === "resources".toUpperCase()) {
-
-                for (var i = 0, il = keyNode.properties.length; i < il; i++) {
-                    if (keyNode.properties[i].name._value.toUpperCase() === "type".toUpperCase()) {
-                        let resourceType = keyNode.properties[i].value._value.toUpperCase();
+            if (rootNode.toString().toUpperCase() === "resources".toUpperCase() && keyOrResourceNode instanceof Json.ObjectValue) {
+                for (var i = 0, il = keyOrResourceNode.properties.length; i < il; i++) {
+                    if (keyOrResourceNode.properties[i].name.toString().toUpperCase() === "type".toUpperCase()) {
+                        let resourceType = keyOrResourceNode.properties[i].value.toString().toUpperCase();
```

Comment:
I see a use case for `for (let property of keyOrResourceNodes.properties)`. I expect tslint [prefer-for-of](https://palantir.github.io/tslint/rules/prefer-for-of/) to complain  about this sometime in the future.

Reply:
NMC